python/langsmith/evaluation/_runner.py

```diff
@@ -1625,23 +1637,43 @@ def _get_run(r: rt.RunTree) -> None:
 
     with rh.tracing_context(enabled=True):
         try:
-            fn(
-                example.inputs,
-                langsmith_extra=rh.LangSmithExtra(
-                    reference_example_id=example.id,
-                    on_end=_get_run,
-                    project_name=experiment_name,
-                    metadata={
-                        **metadata,
-                        "example_version": (
-                            example.modified_at.isoformat()
-                            if example.modified_at
-                            else example.created_at.isoformat()
-                        ),
-                    },
-                    client=client,
-                ),
-            )
+            if with_attachments == True:
+                fn(
+                    example.inputs,
+                    example.attachment_urls,
+                    langsmith_extra=rh.LangSmithExtra(
+                        reference_example_id=example.id,
+                        on_end=_get_run,
+                        project_name=experiment_name,
+                        metadata={
+                            **metadata,
+                            "example_version": (
+                                example.modified_at.isoformat()
+                                if example.modified_at
+                                else example.created_at.isoformat()
+                            ),
+                        },
+                        client=client,
+                    ),
+                )
```

Comment:
what happens if `with_attachment` is `True` but the caller sends in a function that doesn't take in the inputs and attachments as arguments? There needs to be some validation as high up into the call path as possible to fail fast.